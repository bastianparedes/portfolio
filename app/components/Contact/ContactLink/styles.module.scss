@import 'bastianparedes/styles/_breakpoints';

.link {
  width: 100%;
  margin: 2rem 0;
  font-size: 2rem;
  text-align: center;

  &__a {
    width: 100%;
    position: relative;
    text-align: center;
    display: flex;
    text-decoration: none;
    color: inherit;
    transition: all ease 0.2s;

    &:hover {
      bottom: 0.2rem;
      color: rgb(0, 255, 255);
    }
  }
}

@include phone-and-tablet {
  .link {
    margin: 0;
    text-align: center;
    font-size: 4rem;
  }
}

@include phone {
  .link {
    margin: 0;
    text-align: center;
    font-size: 3rem;
  }
}
