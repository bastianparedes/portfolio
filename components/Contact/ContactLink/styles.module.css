.link {
  width: 100%;
  margin: 2rem 0;
  font-size: 2rem;
  text-align: center;
}

.link__a {
  width: 100%;
  position: relative;
  text-align: center;
  display: flex;
  text-decoration: none;
  color: inherit;
  transition: all ease 0.2s;
}

.link__a:hover {
  bottom: 0.2rem;
  color: rgb(0, 255, 255);
}

@media (max-device-width: 700px) {
  .link {
    margin: 0;
    text-align: center;
    font-size: 4rem;
  }
}
