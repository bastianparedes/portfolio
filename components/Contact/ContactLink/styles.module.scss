.link {
  width: 100%;
  margin: 2rem 0;
  font-size: 2rem;
  text-align: center;
}

.link__a {
  width: 100%;
  position: relative;
  text-align: center;
  display: flex;
  text-decoration: none;
  color: inherit;
  transition: all ease 0.2s;

  &:hover {
    bottom: 0.2rem;
    color: rgb(0, 255, 255);
  }
}

@media screen and (max-width: 768px) {
  .link {
    margin: 0;
    text-align: center;
    font-size: 4rem;
  }
}

@media screen and (max-width: 480px) {
  .link {
    margin: 0;
    text-align: center;
    font-size: 3rem;
  }
}
