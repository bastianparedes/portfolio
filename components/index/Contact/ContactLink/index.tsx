import React from 'react';

import styles from './styles.module.scss';

interface LinkProps {
  link: string;
  children: React.ReactNode;
}

const ContactLink = ({ link, children }: LinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a
        className={styles.link__a}
        href={link}
        target="_blank"
        rel="noreferrer noopener"
      >
        {children}
      </a>
    </li>
  );
};

export default ContactLink;
