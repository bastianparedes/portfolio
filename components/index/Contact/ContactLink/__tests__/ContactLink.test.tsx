import React from 'react';

import { render } from '@testing-library/react';

import ContactLink from '..';

describe('<ContactLink />', () => {
  const props = {
    link: 'https://github.com/BastianParedes'
  };

  it('should render', () => {
    const { container } = render(
      <ContactLink {...props}>
        <></>
      </ContactLink>
    );
    expect(container).toMatchSnapshot();
  });
});
