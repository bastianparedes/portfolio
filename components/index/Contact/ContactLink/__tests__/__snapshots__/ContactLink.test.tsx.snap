// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactLink /> should render 1`] = `
<div>
  <li
    class="link"
  >
    <a
      class="link__a"
      href="https://github.com/BastianParedes"
      rel="noreferrer noopener"
      target="_blank"
    />
  </li>
</div>
`;
